scripts/deploy.bash

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+# Deploy assets to DEPLOY_HOST.
+
+# Required environment variables:
+#   DEPLOY_KEY
+#   DEPLOY_USER
+#   DEPLOY_HOST
+#   DEPLOY_PATH
+#   DEPLOY_PORT
+
+# Halt on errors:
+set -e
+# Echo commands:
+set -x
+
+# SSH identity_file (DO NOT inadvertently rsync to production!)
+ID_FILE="ssh_deploy_key"
+
+KH_FILE="known_hosts"
+
+# Protect this private key file:
+umask 077
+# Hide only this command:
+set +x
+echo "${DEPLOY_KEY}" > ${ID_FILE}
+set -x
+
+ssh-keyscan -p ${DEPLOY_PORT} ${DEPLOY_HOST} > ${KH_FILE}
```

Comment:
since we are checking an external resource, check return code
if [ "$?" != '0' ]; 
   echo "failed to get host key from deploy host"
   exit 1
fi

Reply:
ah didn't see the set -e